src/nl/hannahsten/texifyidea/psi/LatexEnvironmentUtil.kt

```diff
@@ -23,7 +25,16 @@ fun getLabel(element: LatexEnvironment): String? {
         optionalParameters.getOrDefault("label", null)
     }
     else {
-        if (!EnvironmentMagic.labeled.containsKey(element.environmentName)) return null
+        // Not very clean. We don't really need the conventions here, but determine which environments *can* have a
+        // label. However, if we didn't use the conventions, we would have to duplicate the information in
+        // EnvironmentMagic
```

Comment:
The information wouldn't be duplicated, if everybody uses it from Magic.
What I did in a similar situation, is I kept the 'hardcoded' magic list, like the one that is now in TexifyConventionsScheme, in the Magic file. This ensures that everything magic is kept in Magic instead of spread around in various managers, while it also allows any code to access the hardcoded list instead of the user-changed list, which is useful if getting the second one has a significant performance impact.
I don't think there is any performance impact in this case, so I think this code is fine as it is.